e2e/nomostest/wait_for_sync.go

```diff
@@ -171,50 +172,54 @@ func (nt *NT) WatchForAllSyncs(options ...WatchForAllSyncsOptions) error {
 	tg := taskgroup.New()
 
 	if opts.syncRootRepos {
-		for name := range nt.RootRepos {
-			if _, ok := opts.skipRootRepos[name]; ok {
+		for id, source := range nt.SyncSources.RootSyncs() {
+			if _, ok := opts.skipRootRepos[id.Name]; ok {
 				continue
 			}
-			nn := RootSyncNN(name)
-			syncDir := syncDirectory(opts.syncDirectoryMap, nn)
+			// TODO: Move SyncPath logic into a SyncSource method
```

Comment:
Can this be done now?

Reply:
Would rather do it in a followup PR. I';m not sure what all it requires changing and it'll be hard to see the diff if added to the same commit.

Reply:
https://github.com/GoogleContainerTools/kpt-config-sync/pull/1406